New recommendation equation

bobbysue opened the pull request:
Changed the recommendation equation based on Dash2in1's comment at https://www.reddit.com/r/AdventureCapitalist/comments/3ik18l/another_calculator/

Looked good to me and I wanted to use it myself, so I changed it.

Slimmmo commented:
Two things.

1. Why leave in ' / (tempPlanet.totalMoneyPerSecond * (tempUnlock / loc.totalMoneyPerSecond)' when the maths says it's supposed to just be '/ (tempPlanet.totalMoneyPerSecond * tempUnlock)'? Were you running into the Javascript limit of ~1.8e+308? I guess it still works since all it does is scale the upgrade scores instead of changing the order of them, but just wondering.

2. Please remind me what was wrong with the Carwash next upgrade?

After that I'll be happy to merge.

bobbysue commented:
1. I originally took out loc.totalMoneyPerSecond, but this gave very low numbers, which showed up as "0".  So I added it back in, and added the constant factor.  I tested at a couple different stages in the games and it gave readable scores, but I didn't do extensive testing.

2.  The carwash wouldn't give a recommendation for the next unlock level, but instead for some other arbitrary increment.  Because of how adding recommendations was handled for newspapers, after the extra recommendations were removed the next increment had to be recalculated.
